src/model.py

```diff
@@ -0,0 +1,54 @@
+# src/model.py
+
+import logging
+from sklearn.naive_bayes import MultinomialNB
+from sklearn.model_selection import train_test_split
+from sklearn.multiclass import OneVsRestClassifier
+from imblearn.pipeline import Pipeline
+from imblearn.over_sampling import RandomOverSampler
+
+logging.getLogger(__name__)
+
+
+def train_model(features, labels, model_config: dict):
+    """
+    Train a model based on the provided configuration.
+
+    Args:
+        features: Feature matrix.
+        labels: Target labels.
+        model_config (dict): Configuration dictionary for the model.
+
+    Returns:
+        Model: Trained model.
+    """
+    type = model_config.get("type")
+    alpha = model_config.get("alpha")
+    oversampling = model_config.get("oversampling", False)
+
+    logging.info("Splitting data into training and testing sets.")
+    x_train, _, y_train, _ = train_test_split(
+        features, labels, test_size=0.2, random_state=42
+    )
+    if type == "naive_bayes":
```

Comment:
If we create a class for each approach/model, we could move the train logic to that class and just call it in here.